### PR TITLE
initial add of TruLens RAG evaluation framework

### PR DESCRIPTION
**Pull request by epinzur**
This contains the primary scripts I've been using for running medium-scale evaluation of different RAG techniques.

A following PR will contain the results from my first test.

### CODE REVIEW

File: evaluation/tru_shared.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            model="text-embedding-ada-002",
+            api_key=os.getenv("AZURE_OPENAI_API_KEY"),
+            api_version="2023-05-15",
+            temperature=temperature,
+        )
+    else:
+        raise Exception(f"Unknown framework: {framework} specified for getEmbeddingsModel()")
+
+
+def getAstraVectorStore(framework: Framework, collection_name: str):
```
Comment (jordanrfrazier) [inline evaluation/tru_shared.py:142]: python convention is `get_astra_vector_store`: https://www.techversantinfotech.com/python-naming-conventions-points-you-should-know/

File: evaluation/tru_shared.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+# OPENAI_API_VERSION
+# TRULENS_DB_CONN_STRING
+
+load_dotenv()
+
+temperature = 0
+
+class Framework(Enum):
+    LANG_CHAIN = "langChain"
+    LLAMA_INDEX = "llamaIndex"
```
Comment (jordanrfrazier) [inline evaluation/tru_shared.py:36]: don't technically have to assign values

File: evaluation/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
+# Evaluation Metrics
```
Comment (jordanrfrazier) [inline evaluation/README.md:1]: Would add instructions on how to run in readme

File: evaluation/tru_shared.py (hunk 1 above)
Comment (epinzur) [reply evaluation/tru_shared.py]: updated all method names in `tru_shared.py`

File: evaluation/README.md (hunk 3 above)
Comment (epinzur) [reply evaluation/README.md:1]: I'm going to hold off on doing this until we make a decision if we are going to publicize TruLens or not. For now I'm the only one running these and I know how to do so.

**Pull request merged**